modules/components/pages/inputs/pg_stream.adoc

```diff
@@ -0,0 +1,414 @@
+
+==== Self-Hosted PostgreSQL
+
+Use an account with sufficient permissions (superuser) to update your replication settings.
+
+. Open the `postgresql.conf` file.
+. Find the `wal_level` parameter.
+. Update the parameter value to `wal_level = logical`. If you already use replication slots, you may need to increase the limit on replication slots (`max_replication_slots`). The `max_wal_senders` parameter value must also be greater than or equal to `max_replication_slots`.
+. Restart the PostgreSQL server.
+
+For this input to make a successful connection to your database, also make sure that your database allows replication connections.
+
+. Open the `pg_hba.conf` file.
+. Update this line.
+
++
+[source,yaml]
+----
+host    replication     <replication-username>   <connector-ip>/32    md5
+----
+
++
+Replace the following placeholders with your own values:
+
+- `<replication-username>`: The username from an account with superuser privileges.
+- `<connector-ip>`: The IP address of the server where you are running Redpanda Connect.
+
+==== Other platforms
+
+- https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.FeatureSupport.LogicalReplication.html[Amazon RDS for PostgreSQL DB^]
+- https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-logical#prerequisites-for-logical-replication-and-logical-decoding[Azure Database for PostgreSQL^]
+- https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication[Google Cloud SQL for PostgreSQL^], including creating a user with replication privileges
+- https://neon.tech/docs/guides/logical-replication-guide[Neon^]
+
+=== Replication process
+
+When you run a pipeline that uses the `pg_stream` input, Redpanda Connect makes a connection to your PostgreSQL database and creates a replication slot. The replication slot uses a copy of the Write-Ahead Log (WAL) file to subscribe to changes in your database records as they are applied to the database itself.
+
+Once the replication slot is available, Redpanda Connect:
+
+. Takes a snapshot of the current database.
+. Starts to stream data updates. It processes messages from the latest offset.
+
+[NOTE]
+====
+
+You can configure this input to run in one of two modes, to either:
+
+- Stream all data in the snapshot and data updates after the snapshot is taken
+- Stream just the data updates
+
+For full details, see the <<stream-snapshot,`stream_snapshot` field>>.
+====
+
+=== Monitoring the replication process
+
+You can monitor the initial replication of data using the following metrics:
+
+- `replication_lag_bytes`: Indicates how far the connector is lagging behind the source database when processing the transaction log.
+- `postgres_snapshot_progress`: Shows the progress of snapshot processing for each table.
+
+Once the initial replication process is complete, the snapshot is removed and the input keeps a connection open to the database so that it can receive data updates.
+
```

Comment:
@le-vlad can we also make sure we document how to receive unchanged TOAST values in updates? I think you mentioned this is something about full replica or some other setting?

Reply:
Added new section.